pfspinup/pfmetadata.py

```diff
@@ -40,22 +39,29 @@ def __init__(self, filename):
         self.folder = os.path.abspath(os.path.dirname(filename))
         self.config = json.loads(open(filename, 'r').read())
 
+        # We initialize the 'mask' attribute (an nz-by-ny-by-nx ndarray) early on,
+        # since this is used in quite a few places later.
+        self.mask = self.input_data('mask', apply_mask=False)
+
     def __getitem__(self, item):
         value = self.config['inputs']['configuration']['data'][item]
         return self._parse_value(value)
 
     def _get_absolute_path(self, filename):
         return os.path.join(self.folder, filename)
 
-    def pfb_data(self, filename):
-        return pfio.pfread(self._get_absolute_path(filename))
+    def pfb_data(self, filename, apply_mask=True):
+        data = pfio.pfread(self._get_absolute_path(filename))
+        if apply_mask:
+            data[self.mask == 0] = np.nan
+        return data
 
-    def input_data(self, which):
+    def input_data(self, which, apply_mask=True):
```

Comment:
By having the metadata object apply the mask when the user script asks for certain input data (pressure, saturation etc.), we avoid the issue of the script having to worry about the mask.